### PR TITLE
Feature/56 implement users list at moderator page

### PR DESCRIPTION
**Pull request by martyniukroman**

### CODE REVIEW

File: src/app/Components/adminpanel/user-management/user-management.component.ts [hunk 1]
```diff
@@ -1,15 +1,128 @@
...
+      this.dataSource = this._userListService.StatusConverter(result.users);
+      this.usersAmount = result.count;
+      this.isLoadingResults = false;
+     },
+      error => {
+      this._notificationService.error(error);
+     });
 
-  ngOnInit() {
+      // this._userListService.getUserList().subscribe((types: any) => {
```
Comment (TREN-D) [inline src/app/Components/adminpanel/user-management/user-management.component.ts]: Remove comments

File: src/app/Components/adminpanel/user-management/user-management.component.ts [hunk 2]
```diff
@@ -1,15 +1,128 @@
...
+  onSearch() {
+    this.paginator.firstPage();
+    const event = new PageEvent();
+    event.pageSize = this._paginationService.userPageSize;
+    event.pageIndex = this._paginationService.pageIndex - 1;
+    event.length = this.usersAmount;
+    this.pageSwitch(event);
   }
 
+  // searchClick() {
```
Comment (TREN-D) [inline src/app/Components/adminpanel/user-management/user-management.component.ts]: here too

File: src/app/Components/adminpanel/user-management/user-management.component.ts [hunk 3]
```diff
@@ -1,15 +1,128 @@
...
+      this.dataSource = result.users;
+      this.usersAmount = result.count;
+    },
+    error => {
+      this._notificationService.error(error);
+    });
+    window.scroll(0, 0);
+  }
+
+  // uniq(Users: UserListModel[]) {
```
Comment (TREN-D) [inline src/app/Components/adminpanel/user-management/user-management.component.ts]: here too

File: src/app/Components/adminpanel/user-management/user-management.component.ts [hunk 4]
```diff
@@ -1,15 +1,128 @@
...
+  // }
+
+  applyFilter(filterValue: any) {
+    this.searchInput = filterValue;
+    filterValue = filterValue.trim();
+    filterValue = filterValue.toLowerCase();
+    this.dataSource.filter = filterValue;
+    this.dataSource.sort = this.sort;
+  }
+  // tslint:disable-next-line:use-life-cycle-interface
```
Comment (TREN-D) [inline src/app/Components/adminpanel/user-management/user-management.component.ts]: here too

File: src/app/Components/adminpanel/user-management/userlist-search/userlist-search.component.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<!-- <div class="example-card example-half-width">
```
Comment (TREN-D) [inline src/app/Components/adminpanel/user-management/userlist-search/userlist-search.component.html:1]: Why all file commented?

File: src/app/Services/UserListService/user-list.service.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  constructor(private http: HttpClient, private paginationService: PaginationService) { }
+
+  getUserList() {
+    const typeUrl = this.url + 'userlist/getparams';
+    return this.http.get(typeUrl, this.httpOptions);
+  }
+  getUserListParams(filter: UserListFilter) {
+    const typeUrl = this.url + 'userlist/getparams';
+
+    // if (filter.searchKey != null && filter.searchKey !== '' ) {
```
Comment (TREN-D) [inline src/app/Services/UserListService/user-list.service.ts]: Remove comments

File: src/app/Services/UserListService/user-list.service.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    return this.http.get(typeUrl, this.httpOptions);
+  }
+
+  StatusConverter(Users: any) {
+
+    console.log(Users);
+
+    Users.forEach(element => {
+      if (element.statusId === 1 || element.statusName === 'Banned') {
+        element.statusId = 1;
```
Comment (TREN-D) [inline src/app/Services/UserListService/user-list.service.ts:43]: Whats the logic here? You are checking if its equal 1 and then set it to 1..

Comment (martyniukroman) [reply src/app/Services/UserListService/user-list.service.ts:43]: in future we can post or put user, and this will be easier fill

**Review by TREN-D — APPROVED** [review]

**Review by SergiyKozachok — APPROVED** [review]

**Review by eugent482 — APPROVED** [review]

**Pull request merged**